crates/db_schema/src/impls/images.rs

```diff
@@ -80,6 +82,17 @@ impl LocalImage {
       .execute(conn)
       .await
   }
+
+  pub async fn delete_by_url(pool: &mut DbPool<'_>, url: &DbUrl) -> Result<Self, Error> {
+    let conn = &mut get_conn(pool).await?;
+    static IMAGE_REGEX: Lazy<Regex> =
+      Lazy::new(|| Regex::new(r"^.*/pictrs/image/([a-z0-9-]+\.[a-z]+)$").expect("compile regex"));
```

Comment:
This isnt very elegant but I dont see any better way unless we separately store the alias for each avatar image.

Reply:
Regex is overkill here IMO. 

A pictrs avatar will always be at something like: `https://lemmy.world/pictrs/image/78b09748-85b9-4678-8375-096b57595903.jpeg` . So just split by `/`, pop the last one, and remove the extension to get the alias.

I'd also recommend this function not running the diesel::delete, but just extracting the alias, and running `Self::delete_by_alias`

Reply:
Makes sense, fixed.